src/GHC/Util/HsExpr.hs

```diff
@@ -290,13 +292,21 @@ descendBracketOld op x = (descendIndex g1 x, descendIndex g2 x)
     g1 = (fst .) . g
     g2 = (snd .) . g
 
-    f i (L _ (HsPar _ y)) z | not $ needBracketOld i x y = (y, z)
-    f i y z                 | needBracketOld i x y = (addParen y, addParen z)
+    f i (L _ (HsPar _ y)) z
+      | not $ needBracketOld i x y = (y, z)
+    f i y z
+      | needBracketOld i x y = (addParen y, addParen z)
+      -- https://github.com/mpickering/apply-refact/issues/7
+      | isOp y = (y, addParen z)
     f _ y z                 = (y, z)
 
     f1 = ((fst .) .) . f
     f2 = ((snd .) .) . f
 
+    isOp = \case
+      L _ (HsVar _ var) -> (not . all isAlphaNum) (rdrNameStr var)
```

Comment:
I bet this classifies `foo'` as an operator, which I imagine is wrong? And probably `_foo` too?

Reply:
I see `isOp` in src\Hint\Extensions.hs that looks better. Perhaps put that somewhere general and reuse in both cases?

Reply:
This one?
```
    isOp :: LIdP GhcPs -> Bool
    isOp name = case rdrNameStr name of
      (c:_) -> not $ isAlpha c || c == '_'
      _ -> False
```
> Perhaps put that somewhere general and reuse in both cases?

Pull it down to `ghc-lib-parser-ex` perhaps?

Reply:
In any case, it's got an interesting signature. It's not wrong but I more obviously expressed as `isOp :: Located RdrName -> Bool`.

Reply:
Why not just RdrName? And does GHC itself already have something?

Reply:
> Why not just RdrName?

That'd work.

> And does GHC itself already have something?

That's an excellent question! `ghc-exactprint` offers this:
```haskell
isSymbolRdrName :: GHC.RdrName -> Bool
isSymbolRdrName n = GHC.isSymOcc $ GHC.rdrNameOcc n
```
where,
```haskell
isSymOcc :: Name -> Bool
isSymOcc n
  = case nameBase n of
      []    -> True  -- Empty name; weird
      (c:_) -> startsVarSym c
                   -- c.f. OccName.startsVarSym in GHC itself
```
So, what do you think? Shall we harvest the ghc-exactprint definition?

Reply:
TIL
```haskell

-----------------------------------------------------------------------------
-- |
-- Module      :  GHC.Lexeme
-- Copyright   :  (c) The GHC Team
--
-- Maintainer  :  ghc-devs@haskell.org
-- Portability :  portable
--
-- Functions to evaluate whether or not a string is a valid identifier.
--
module GHC.Lexeme (
          -- * Lexical characteristics of Haskell names
        startsVarSym, startsVarId, startsConSym, startsConId,
        startsVarSymASCII, isVarSymChar, okSymChar
  ) where
```
Nice!

Reply:
Fyi, `isSymOcc` (via `okSymChar` via `startsVarSym`) correctly rules out `_foo`.